JK Hera/src/main/java/ayds/hera/songinfo/moredetails/view/ArticleDescriptionHelper.kt

```diff
@@ -0,0 +1,39 @@
+package ayds.hera.songinfo.moredetails.view
+
+import ayds.hera.songinfo.moredetails.model.entities.Article
+import ayds.hera.songinfo.moredetails.model.entities.EmptyArticle
+import ayds.hera.songinfo.moredetails.model.entities.NYTArticle
+
+
+interface ArticleDescriptionHelper {
+    fun getArticleDescriptionText(artistName: String, article: Article = EmptyArticle): String
+}
+private const val LOCAL_MARK = "[*]"
+
+internal class ArticleDescriptionHelperImpl : ArticleDescriptionHelper {
+    override fun getArticleDescriptionText(artistName: String, article: Article): String {
+        return when(article) {
+            is NYTArticle -> textToHtml(artistName, article)
+            else -> "No hay descripción"
+        }
+    }
+
+    private fun textToHtml(artistName: String, article: Article): String {
+        val locallyStoraged = if (article.isLocallyStoraged) LOCAL_MARK else ""
+
+        val builder = StringBuilder()
+        builder.append("<html><div width=400 >")
+        builder.append("<font face=\"arial\">")
+        builder.append(locallyStoraged)
+
+        val textWithBold = article.articleData
+            .replace("'", " ")
+            .replace("\n", "<br>")
+        .replace("(?i)" + artistName.toRegex(), "<b>" + artistName.toUpperCase() + "</b>")
+
+        builder.append(textWithBold)
+        builder.append("</font></div></html>")
+
+        return builder.toString()
```

Comment:
Se puede simplificar: 
```suggestion
        return StringBuilder().apply {
           append("<html><div width=400 >")
           append("<font face=\"arial\">")
           append(locallyStoraged())
           append(articleData() )         
           append("</font></div></html>")
        }.toString()       
```